scraibe/diarisation.py

```diff
@@ -211,64 +212,65 @@ def load_model(cls,
+                pwd = "/".join(pwd)
+
+                path_to_model = os.path.join(pwd, "pytorch_model.bin")
 
                 if not os.path.exists(path_to_model):
-                    warnings.warn(f"Model not found at {path_to_model}. "\
-                        "Trying to find it nearby the config file.")
-                    
-                    pwd = model.split("/")[:-1]
-                    pwd = "/".join(pwd)
-                    
-                    path_to_model = os.path.join(pwd, "pytorch_model.bin")
-
-                    if not os.path.exists(path_to_model):
-                        warnings.warn(f"Model not found at {path_to_model}. \
-                            'Trying to find it nearby .bin files instead.")
-                        # list elementes with the ending .bin
-                        bin_files = [f for f in os.listdir(pwd) if f.endswith(".bin")]
-                        if len(bin_files) == 1:
-                            path_to_model = os.path.join(pwd, bin_files[0])
-                        else:
-                            warnings.warn("Found more than one .bin file. "\
-                                "or none. Please specify the path to the model " \
-                                "or setup a huggingface token.")
-                                                
-                    warnings.warn(f"Found model at {path_to_model} overwriting config file.")
-                    
-                    config['pipeline']['params']['segmentation'] = path_to_model
-                    
-                    with open(model, 'w') as file:
-                        yaml.dump(config, file)
-                                
-            _model =  Pipeline.from_pretrained(model,
-                                            use_auth_token = use_auth_token,
-                                            cache_dir = cache_dir,
-                                            hparams_file = hparams_file,)
+                    warnings.warn(f"Model not found at {path_to_model}. \
+                        'Trying to find it nearby .bin files instead.")
+                    # list elementes with the ending .bin
+                    bin_files = [f for f in os.listdir(pwd) if f.endswith(".bin")]
+                    if len(bin_files) == 1:
+                        path_to_model = os.path.join(pwd, bin_files[0])
+                    else:
+                        warnings.warn("Found more than one .bin file. "\
+                            "or none. Please specify the path to the model " \
+                            "or setup a huggingface token.")
+                        raise FileNotFoundError
+
+                warnings.warn(f"Found model at {path_to_model} overwriting config file.")
+
+                config['pipeline']['params']['segmentation'] = path_to_model
+
+                with open(model, 'w') as file:
+                    yaml.dump(config, file)
+        elif isinstance(model, tuple):
```

Comment:
Why tuple?

Reply:
Because the non-yaml default is a tuple of the two hf repos. It's a bit ugly to get them checked with different auth token, but it works so far.